sys/include/saul_reg.h

```diff
@@ -84,56 +99,56 @@ int saul_reg_rm(saul_reg_t *dev);
  *
  * @param[in] pos       position to look up
  *
- * @return      pointer to the device at position specified by @p pos
- * @return      NULL if no device is registered at that position
+ * @return      complex pointer to the device at position specified by @p pos
+ * @return      .real = NULL if no device is registered at that position
  */
-saul_reg_t *saul_reg_find_nth(int pos);
+saul_complex_ptr_t saul_reg_find_nth(int pos);
```

Comment:
I would change this to
`int saul_complex_ptr_t saul_reg_find_nth(saul_complex_ptr_t *node, int pos);`
and have it return `0` or `1` based on if there is a device at the position. This way checks if the device at `@p pos` exists are still easy and you have only an extra pointer in the arguments.

Reply:
```saul_reg_find_nth(pos).real != NULL``` gives you an easy way to test this as well. If you want to assign and test at the same time you could even do ```(foo = saul_reg_find_nth(pos)).real != NULL```. Both are similar to the previous behavior (with just the addition of .real).

Reply:
Problem is that this will require a double pointer to be passed in the argument, because you want saul_reg_find_nth to be able to modify it rather than its own local copy.